src/AltinnCore/Designer/Controllers/DeployController.cs

```diff
@@ -233,6 +233,15 @@ private async Task<bool> RegisterApplicationInStorage(string applicationOwnerId,
         {
             bool applicationInStorage = false;
             ApplicationMetadata applicationMetadataFromRepository = _repository.GetApplicationMetadata(applicationOwnerId, applicationCode);
+
+            // for old service application meta data file was not generated, so create the application meta data file
+            // but the metadata for attachment will not be available on deployment
+            if (applicationMetadataFromRepository == null)
```

Comment:
Should this functionality be marked for future removal ? Once we deploy a breaking change and all services must be recreated the need for this will be obsolete.

Reply:
Yes, and it's also not possible to filter out the metadata information for attachment on deploy.